File: TouchNothing/Utilities/PinInput.swift
import Foundation

enum PinInput {
    static func sanitize(_ value: String) -> String {
        String(value.filter(\.isNumber).prefix(AppConstants.pinLength))
    }

    static func isValid(_ value: String) -> Bool {
        value.count == AppConstants.pinLength && value.allSatisfy(\.isNumber)
    }

    static func randomPin() -> String {
        String(format: "%04d", Int.random(in: 0...9_999))
    }
}

File: app/src/main/java/com/touchnothing/app/util/PinInput.kt
package com.touchnothing.app.util

object PinInput {
    fun sanitize(value: String): String =
        value.filter { it.isDigit() }.take(AppConstants.PIN_LENGTH)

    fun isValid(value: String): Boolean =
        value.length == AppConstants.PIN_LENGTH && value.all { it.isDigit() }

    fun randomPin(): String = "%04d".format((0..9999).random())
}
